core/ui/SideBarSegments/search.tid

```diff
@@ -4,23 +4,27 @@ tags: $:/tags/SideBarSegment
 \whitespace trim
 <div class="tc-sidebar-lists tc-sidebar-search">
 
-<$set name="searchTiddler" value="$:/temp/search">
+<$vars searchTiddler="$:/temp/search/input" searchListState=<<qualify "$:/state/search-list/selected-item">> filter1="[!is[system]search:title<userInput>sort[title]limit[250]]" filter2="[!is[system]search<userInput>sort[title]limit[250]]">
 <div class="tc-search">
-<$edit-text tiddler="$:/temp/search" type="search" tag="input" focus={{$:/config/Search/AutoFocus}} focusPopup=<<qualify "$:/state/popup/search-dropdown">> class="tc-popup-handle"/>
-<$reveal state="$:/temp/search" type="nomatch" text="">
+<$macrocall $name="keyboard-driven-input" tiddler="$:/temp/search" type="search" tag="input" focus={{$:/config/Search/AutoFocus}} focusPopup=<<qualify "$:/state/popup/search-dropdown">> class="tc-popup-handle" inputactions="""<$action-setfield $tiddler="$:/temp/search/input" text={{$:/temp/search}}/>""" storeTitle="$:/temp/search/input" filter1=<<filter1>> filter2=<<filter2>> input-cancel-actions="""<$action-deletetiddler $tiddler="$:/temp/search"/><$action-deletetiddler $tiddler="$:/temp/search/input"/><$action-deletetiddler $tiddler=<<selectionTiddler>>/>""" input-accept-actions="""<$action-navigate $to={{{ [<__tiddler__>get[text]] }}}/>""" input-accept-variant-actions="""<$action-sendmessage $message="tm-edit-tiddler" $param={{{  [<__tiddler__>get[text]] }}}/>""" filterMinLength={{$:/config/Search/MinLength}} refreshTitle="$:/temp/search/refresh" selectionTiddler=<<searchListState>>/>
+<$reveal state="$:/temp/search/input" type="nomatch" text="">
 <$button tooltip={{$:/language/Buttons/AdvancedSearch/Hint}} aria-label={{$:/language/Buttons/AdvancedSearch/Caption}} class="tc-btn-invisible">
 <$action-setfield $tiddler="$:/temp/advancedsearch" text={{$:/temp/search}}/>
 <$action-setfield $tiddler="$:/temp/search" text=""/>
+<$action-setfield $tiddler="$:/temp/search/input" text=""/>
+<$action-setfield $tiddler=<<searchListState>> text=""/>
```

Comment:
Can we just delete the $:/temp/search, $:/temp/search/input and searchListState instead?

We could then replace those 3 action-setfield widgets with an action-deletetiddler widget? Untested code below:
`<$action-deletetiddler $filter=""""[[$:/temp/search]] [[$:/temp/search/input]] [<searchListState>]"""`

Reply:
Hi @saqimtiaz , yes that should work! I'll test it

Reply:
I believe there are more like these somewhere, I'll investigate